RELEASING.md

```diff
@@ -0,0 +1,22 @@
+# Releasing `pywatershed`
```

Comment:
I don't think this should be in the root directory. Having it within `ci/` or in `github/` is probably better since this is only something that maintainers will be doing.

Also `RELEASE.md` is probably a better name.

Reply:
moved to `.github/RELEASE.md`